pkg/sfu/downtrack.go

```diff
@@ -540,82 +544,27 @@ func (d *DownTrack) MaxTemporalLayer() int32 {
 	return d.maxTemporalLayer.get()
 }
 
-// switchSpatialLayer switches the current layer
-func (d *DownTrack) switchSpatialLayer(targetLayer int32) error {
-	if d.trackType != SimulcastDownTrack {
-		return ErrSpatialNotSupported
-	}
-
-	// already set
-	if d.CurrentSpatialLayer() == targetLayer {
-		return nil
-	}
-
+// switchSpatialLayer switches the target layer
+func (d *DownTrack) switchSpatialLayer(targetLayer int32) {
 	d.targetSpatialLayer.set(targetLayer)
-	return nil
 }
 
-func (d *DownTrack) UptrackLayersChange(availableLayers []uint16, layerAdded bool) (int32, error) {
-	if d.trackType == SimulcastDownTrack {
-		currentLayer := uint16(d.CurrentSpatialLayer())
-		maxLayer := uint16(d.maxSpatialLayer.get())
-
-		var maxFound uint16 = 0
-		layerFound := false
-		var minFound uint16 = 0
-		for _, target := range availableLayers {
-			if target <= maxLayer {
-				if target > maxFound {
-					maxFound = target
-					layerFound = true
-				}
-			} else {
-				if minFound > target {
-					minFound = target
-				}
-			}
-		}
-		var targetLayer uint16
-		if layerFound {
-			targetLayer = maxFound
-		} else {
-			targetLayer = minFound
-		}
-		if currentLayer != targetLayer {
-			// LK-TODO-START
-			// This layer switch should be removed when StreamAllocator is used.
-			// Available layers change should be signalled to StreamAllocator
-			// and StreamAllocator will take care of adjusting allocations.
-			// LK-TODO-END
-			if err := d.switchSpatialLayer(int32(targetLayer)); err != nil {
-				return int32(targetLayer), err
-			}
-		}
-
-		if d.onAvailableLayersChanged != nil {
-			d.onAvailableLayersChanged(d, layerAdded)
-		}
-
-		return int32(targetLayer), nil
+func (d *DownTrack) UptrackLayersChange(availableLayers []uint16, layerAdded bool) {
+	if d.onAvailableLayersChanged != nil {
+		d.onAvailableLayersChanged(d, layerAdded)
 	}
-	return -1, fmt.Errorf("downtrack %s does not support simulcast", d.id)
 }
 
 func (d *DownTrack) switchTemporalLayer(targetLayer int32) {
-	if d.trackType != SimulcastDownTrack {
-		return
-	}
-
-	layer := d.temporalLayer.get()
-	currentLayer := uint16(layer)
-	currentTargetLayer := uint16(layer >> 16)
+	d.targetTemporalLayer.set(targetLayer)
+}
 
-	// Don't switch until previous switch is done or canceled
-	if currentLayer != currentTargetLayer {
-		return
-	}
+func (d *DownTrack) disableSend() {
```

Comment:
This is how we disable a bandwidth constrained track now. Set all layers to invalid values.

Reply:
No that has a separate enabled flag